tests/test_backend.py

```diff
@@ -35,56 +33,74 @@ def test_none_flow_graph(self, mock_save_graph):
     @patch('hidet.save_graph')
     def test_exception_on_save_graph_failure(self, mock_save_graph):
         mock_save_graph.side_effect = Exception("Test Exception")
+        flowgraph = MagicMock(spec=hidet.FlowGraph)
+
         with self.assertRaises(Exception) as context:
-            self.runner._get_model_id(self.flow_graph)
+            self.runner._get_model_id(flowgraph)
 
         self.assertIn("Getting model id: failed to save FlowGraph.", str(context.exception))
         mock_save_graph.assert_called_once()
 
-    # Given the same flow graph, the model id should be the same
-    def test_flow_graph_id_consistency(self):
-        model_id1 = self.runner._get_model_id(self.flow_graph)
-        model_id2 = self.runner._get_model_id(self.flow_graph)
-        self.assertEqual(model_id1, model_id2)
-
-    # Given the same torch fx graph, the model id should be the same
-    def test_torch_fx_graph_id_consistency(self):
-        graph_module_2 = deepcopy(self.graph_module)
-        flow_graph_2 = self.graph_module_to_flow_graph(graph_module_2)
-
-        model_id_1 = self.runner._get_model_id(self.flow_graph)
-        model_id_2 = self.runner._get_model_id(flow_graph_2)
-        self.assertEqual(model_id_1, model_id_2)
-
-    # Given the same torch fx graph, the model id should be the same
-    def test_model_id_consistency(self):
-        model_2 = deepcopy(self.model)
-        graph_module_2 = torch.fx.symbolic_trace(model_2)
-        flow_graph_2 = self.graph_module_to_flow_graph(graph_module_2)
-
-        model_id_1 = self.runner._get_model_id(self.flow_graph)
-        model_id_2 = self.runner._get_model_id(flow_graph_2)
-        self.assertEqual(model_id_1, model_id_2)
-
-    # Given two different models, the model id should be different
-    def test_output_uniqueness(self):
-        model_34 = torch.hub.load('pytorch/vision:v0.10.0', 'resnet34', pretrained=True, verbose=False).eval()
-        graph_module_34 = torch.fx.symbolic_trace(model_34)
-        flow_graph_34 = self.graph_module_to_flow_graph(graph_module_34)
-
-        model_id1 = self.runner._get_model_id(self.flow_graph)
-        model_id2 = self.runner._get_model_id(flow_graph_34)
-        self.assertNotEqual(model_id1, model_id2)
+    # Stop remote_compilation after grabbing model_id from _wait_for_status
+    def exit_early(self, *args):
+        raise Exception("Exiting early")
+
+    # Don't run remote_compile in a seperate thread
+    def start_func(self):
+        self._target()
```

Comment:
what does this do?

Reply:
start_func is a function that replaces any calls to threading.Thread.start(). This means that the "self"s in this function refers to the Thread object, not the self of TestGetModelId. With that said, the _target attribute of a Thread object holds the function to be called in a seperate thread. In this case, I am calling the function directly to avoid multithreading.

However I think using self here to refer to the Thread self and not the TestGetModelId self is a bit confusing. I should probably change it to something like self_thread or _self like I did somewhere else